packages/rule-tester/src/RuleTester.ts

```diff
@@ -986,20 +997,43 @@ export class RuleTester extends TestFramework {
 
     if (hasOwnProperty(item, 'output')) {
       if (item.output == null) {
+        if (result.outputs.length) {
+          assert.strictEqual(
+            result.outputs[0],
+            item.code,
+            'Expected no autofixes to be suggested.',
+          );
+        }
+      } else if (typeof item.output === 'string') {
+        assert(result.outputs.length > 0, 'Expected autofix to be suggested.');
         assert.strictEqual(
-          result.output,
-          item.code,
-          'Expected no autofixes to be suggested',
+          result.outputs[0],
+          item.output,
+          'Output is incorrect.',
         );
+        if (result.outputs.length) {
+          assert.deepStrictEqual(
+            result.outputs,
+            [item.output],
+            'Multiple autofixes are required due to overlapping fix ranges - please use the array form of output to declare all of the expected autofix passes.',
+          );
+        }
       } else {
-        assert.strictEqual(result.output, item.output, 'Output is incorrect.');
+        assert(result.outputs.length > 0, 'Expected autofix to be suggested.');
+        assert.deepStrictEqual(
+          result.outputs,
+          item.output,
+          'Outputs do not match.',
+        );
       }
     } else {
-      assert.strictEqual(
-        result.output,
-        item.code,
-        "The rule fixed the code. Please add 'output' property.",
-      );
+      if (result.outputs.length) {
+        assert.strictEqual(
+          result.outputs[0],
+          item.code,
+          "The rule fixed the code. Please add 'output' property.",
```

Comment:
```suggestion
          "The rule fixed the code. Please add 'output' or 'outputs' property.",
```

Reply:
There is only a publicly visible property `output?: string | string[] | null` :smile: so we can't suggest adding `outputs` property. I think the main confusion here is that `runRuleForItem` now returns an array of `outputs` rather than just the string `output`

Reply:
Oh right sorry 🤦 I meant to delete this comment. Ignore me!